senator/game.py

```diff
@@ -11,20 +11,29 @@ def regret_matching(regret: np.ndarray) -> np.ndarray:
 
 
 class Node:
-    def __init__(self, parent, utility_matrix):
+    def __init__(self, parent, utility_matrix: np.ndarray) -> None:
+
         self.parent = parent
         self.utility_matrix = utility_matrix
         self.iterations = 0
 
         self._num_players = utility_matrix.shape[0]
         self._num_resources = utility_matrix.shape[1]
+
         self._cumulative_regret = np.zeros((self._num_players, self._num_resources))
-        self._cumulative_positive_regret = np.zeros((self._num_players, self._num_resources))
-        self.children = {}  # key: action, value: Node
+        self._cumulative_strategy = np.zeros((self._num_players, self._num_resources))
+
+        self.children: dict[tuple, Node] = {}  # key: action, value: Node
 
     @property
     def current_strategy(self) -> np.ndarray:
-        strategy = regret_matching(self._cumulative_positive_regret)
+        strategy = regret_matching(self._cumulative_regret)
+        return strategy
+
+    @property
+    def average_strategy(self) -> np.ndarray:
+        strategy = regret_matching(self._cumulative_strategy)
+
```

Comment:
Tady je podezřelý že máš dvě funkce který vypadají skoro stejně, a jedna z nějakého důvodu magicky dává na místo regretu tu cum strategy.

Zaprvé, @DavidSych je to takhle ok?

Zadruhé, jestli jo, tak bych tu funkci `regret_matching` přejmenoval na něco jako `normalize_to_strategies` nebo něco v tom smyslu aby bylo jasný co dělá, a proč teda je ok dávat tam něco co není regret.

Reply:
Jo ok tak v tom případě bych tu funkci co to volá jen přejmenoval na něco jako `normalize....`. :)